src/shim/kernel.rs

```diff
@@ -8,95 +8,50 @@ use fvm4::kernel::SyscallError as E4;
 use fvm_shared2::error::ErrorNumber as N2;
 use fvm_shared3::error::ErrorNumber as N3;
 use fvm_shared4::error::ErrorNumber as N4;
-use itertools::Either;
+use num_derive::FromPrimitive;
+use num_traits::FromPrimitive;
 use std::fmt;
 
-macro_rules! error_number {
-    ($($variant:ident),* $(,)?) => {
-        #[derive(Debug, Clone)]
-        #[non_exhaustive]
-        pub enum ErrorNumber {
-            $($variant,)*
-            Unknown(UnknownErrorNumber),
-        }
-
-        $(
-            static_assertions::const_assert_eq!(N2::$variant as u32, N3::$variant as u32);
-        )*
-
-        impl From<N2> for ErrorNumber {
-            fn from(value: N2) -> Self {
-                match value {
-                    $(N2::$variant => Self::$variant,)*
-                    u => Self::Unknown(UnknownErrorNumber::N2(u)),
-                }
-            }
-        }
-
-        impl From<N3> for ErrorNumber {
-            fn from(value: N3) -> Self {
-                match value {
-                    $(N3::$variant => Self::$variant,)*
-                    u => Self::Unknown(UnknownErrorNumber::N3(u)),
-                }
-            }
-        }
-
-        impl From<N4> for ErrorNumber {
-            fn from(value: N4) -> Self {
-                match value {
-                    $(N4::$variant => Self::$variant,)*
-                    u => Self::Unknown(UnknownErrorNumber::N4(u)),
-                }
-            }
-        }
-
-        impl ErrorNumber {
-            fn as_unshimmed(&self) -> Either<N3, Either<N4, N2>> {
-                match self {
-                    $(Self::$variant => Either::Left(N3::$variant),)*
-                    Self::Unknown(u) => u.as_unshimmed(),
-                }
-            }
-        }
-    };
+impl From<N2> for ErrorNumber {
+    fn from(value: N2) -> Self {
+        FromPrimitive::from_u32(value as u32).expect("conversion from N2 must work")
```

Comment:
You're not allowed to panic. You may `expect` when you can prove correctness to a human but not to rustc. However, as a human, `conversion from N2 must work` is not convincing to me. Either convince me with a proof of correctness or don't use `expect`.

Reply:
I've added static assertions.

Reply:
I still see `expect`.